### PR TITLE
Fix wrong status code in error monitoring sample solution

### PR DESCRIPTION
**Pull request by mafge**
While migrating the error monitoring sample solution to AWS CDK I figured out that one of the requests in the Lambda function is not returning the expected http status code (instead of 404 the call returned 400). This is fixed with this PR.

Also changed in this PR:
- Add a section to the README to set region code and marketplace id before deploying (otherwise the solution doesn't really work for non US accounts)
- Remove duplicated code from ErrorMonitoringHandler
- Add marketplace ids for multiple regions to the Constants file to make switching them as easy as possible
- Some boyscouting (optimize imports, fix warnings etc.)

### CODE REVIEW

File: use-cases/error-monitoring/README.md [hunk 1]
```diff
@@ -56,7 +56,12 @@ EmailId=user@domain.com
 Schedule=rate(1 minute)
 ```
 
-### 2. Configure Sample Solution App's IAM user
+### 2. Set region code and marketplace id
+You have to set region code and marketplace id that matches your account in the code. By default it is set to US.
+If you want to change the values go to [ErrorMonitoringHandler.java](https://github.com/amzn/selling-partner-api-samples/blob/main/use-cases/error-monitoring/code/java/src/main/java/lambda/ErrorMonitoringHandler.java) and enter the proper data.
```
Comment (rodrifed) [inline use-cases/error-monitoring/README.md:61]: Here you can point to the specific lines: https://github.com/amzn/selling-partner-api-samples/blob/main/use-cases/error-monitoring/code/java/src/main/java/lambda/ErrorMonitoringHandler.java#L28-L31

File: use-cases/error-monitoring/README.md [hunk 2]
```diff
@@ -56,7 +56,12 @@ EmailId=user@domain.com
...
+Available values can be found in [Constants.java](https://github.com/amzn/selling-partner-api-samples/blob/main/use-cases/error-monitoring/code/java/src/main/java/lambda/utils/Constants.java).
```
Comment (rodrifed) [inline use-cases/error-monitoring/README.md:62]: Same as above

**Review by rodrifed — COMMENTED** [review]
Thanks for the proactive approach to fix these issues. I left 2 minor comments.

**Review by rodrifed — APPROVED** [review]

File: use-cases/error-monitoring/code/java/src/main/java/lambda/ErrorMonitoringHandler.java [hunk 3]
```diff
@@ -2,152 +2,131 @@
...
-            List<String> marketplaceIds = Arrays.asList(US_MARKETPLACE_ID);
-
-            ApiResponse<Item> httpResult = listingsApi.getListingsItemWithHttpInfo(sellerId, sku, marketplaceIds, null, includedData);
+            CatalogApi catalogApi = catalogItemsApi(regionCode, refreshToken);
 
-            context.getLogger().log("Status Code: " + httpResult.getStatusCode());
-            context.getLogger().log("Headers: " + httpResult.getHeaders());
-            context.getLogger().log("Data: " + httpResult.getData());
+            ApiResponse<Item> httpResult = catalogApi.getCatalogItemWithHttpInfo(
+                    "0000000001",
```
Comment (MeenaAmz) [inline use-cases/error-monitoring/code/java/src/main/java/lambda/ErrorMonitoringHandler.java:84]: What is this string? if this is a constant, can we define it in constants?

**Comment by MeenaAmz** [comment]
> While migrating the error monitoring sample solution to AWS CDK I figured out that one of the requests in the Lambda function is not returning the expected http status code (instead of 404 the call returned 400). This is fixed with this PR.
> 
> Also changed in this PR:
> 
> * Add a section to the README to set region code and marketplace id before deploying (otherwise the solution doesn't really work for non US accounts)
> * Remove duplicated code from ErrorMonitoringHandler
> * Add marketplace ids for multiple regions to the Constants file to make switching them as easy as possible
> * Some boyscouting (optimize imports, fix warnings etc.)

Changes look good, except for the one comment. Thanks for proactively fixing this, Marc.

**Pull request merged**